### ISSUE TITLE
Is it possible to add my own traineddata?

### ISSUE THREAD

**Issue by marko1943** [jonathanpalma/react-native-tesseract-ocr#13]
Hi,

I was wondering is this possible?
What I though would it be possible edit this java class where you declare languages and trained data
https://github.com/jonathanpalma/react-native-tesseract-ocr/blob/master/android/src/main/java/com/reactlibrary/RNTesseractOcrModule.java

Add some lang of my own, and then reference it in react & use it? Would that work as expected?

**Comment by jonathanpalma** [comment]
Hey @marko1943 
It's a good idea, I'll add a LANG_CUSTOM constant to handle customs traineddata files under the name `custom.traineddata`

**Comment by marko1943** [comment]
@jonathanpalma great stuff, thanks!

**Comment by jonathanpalma** [comment]
You're welcome @marko1943 

You can check it out [here](https://github.com/jonathanpalma/react-native-tesseract-ocr#if-you-want-to-use-your-own-trained-data-file)!! Please let me know if the solution meets the requirements

**Comment by marko1943** [comment]
I believe it should be all good. Can't test right now since I'm still working on my custom trained data but yap, this should do the job.

**Comment by marko1943** [comment]
@jonathanpalma Confirming it works.

One question, I saw it on your to-do and noticed you're planing to implement setVariable from tessBaseApi to implement whitelists/blacklists, how is that going?

I started working on that but I'm just finding my way through all of this, I'm actually coming from Ionic and moved here cause Ionic plugin for tesseract and tess-two is not that good, this one has a better logic and works better -> wish Ionic had a version of this as well lol (with implemented blacklists and whitelists)

**Comment by jonathanpalma** [comment]
@marko1943 Thank you for your support!

I'm looking for the best way to include whitelists and blacklists considering that many people who use [tesseract.js](https://github.com/naptha/tesseract.js) for web & node environment would like to use this for react-native since tesseract.js doesn't support it for now.

```
/**
 * @param {string} imgPath - The path of the image.
 * @param {string} lang - The language you want to process.
 * @param {string} object.blacklist - List of characters you DON'T want to detect
 * @param {string} object.whitelist - List of characters you want to detect
 */
RNTesseractOcr.recognize(imgPath, lang, { blacklist: '', whitelist: '' })
  .then((result) => {
    this.setState({ ocrResult: result });
    console.log("OCR Result: ", result);
  })
  .catch((err) => {
    console.log("OCR Error: ", err);
  })
  .done();
```

[Tesseract.js - react-native support](https://github.com/naptha/tesseract.js/issues/55)

**Comment by marko1943** [comment]
I might be understanding this wrong, but your plugin is actually wrapping the tess-two library, [TessBaseApi](https://github.com/rmtheis/tess-two/blob/master/tess-two/src/com/googlecode/tesseract/android/TessBaseAPI.java)

You're basically using a getUTF8() method to recognized text (the Ionic plugin also does this).

So looking at the TessBaseAPI, I'm not sure I understand you but wouldn't it be the same process to use setVariable method from there for whitelists just like you use getUTF8 method?

**Comment by marko1943** [comment]
by the way - everything was working just fine yesterday, today I keep getting this error
https://github.com/facebook/react-native/issues/11389

tried upgrading your project to 0.40+ but it didn't work and I'm already wasting too much time debugging something that shouldn't even exist in the first place -> it's not your fault it's reacts' lol, it's kinda buggy framework

**Comment by iancanderson** [comment]
@marko1943 Hi, I saw this because you linked to the issue on RN - it's fixed in `0.39.2`, so no need to upgrade to `0.40.0`: https://github.com/facebook/react-native/compare/v0.39.0...v0.39.2

**Comment by marko1943** [comment]
@iancanderson  thanks for the reply, I'm kind of too busy on other things so I've put this on hold.
Well, the version of this [example](https://github.com/jonathanpalma/react-native-tesseract-ocr/blob/master/tesseractOcrSample/package.json) is  "react-native": "^0.33.0" 

the weird thing was that i worked 1 day and the next it wasn't - cant remember doing anything worthy that would make that happen but yeah, kind of busy for now, we'll see what happens when I need this again
